modules/core/src/test/java/org/apache/ignite/internal/processors/cache/transactions/NonTransactionalOperationsInTxTest.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.processors.cache.transactions;
+
+import javax.cache.CacheException;
+import org.apache.ignite.IgniteCache;
+import org.apache.ignite.configuration.IgniteConfiguration;
+import org.apache.ignite.internal.IgniteEx;
+import org.apache.ignite.testframework.junits.common.GridCommonAbstractTest;
+import org.junit.Test;
+
+import static org.apache.ignite.cache.CacheAtomicityMode.TRANSACTIONAL;
+import static org.apache.ignite.testframework.GridTestUtils.assertThrows;
+
+/** Checks that non-transactional cache operations fail within a transaction. */
+public class NonTransactionalOperationsInTxTest extends GridCommonAbstractTest {
+    /** {@inheritDoc} */
+    @Override protected IgniteConfiguration getConfiguration(String gridName) throws Exception {
+        IgniteConfiguration cfg = super.getConfiguration(gridName)
+            .setCacheConfiguration(defaultCacheConfiguration().setAtomicityMode(TRANSACTIONAL));
+
+        return cfg;
+    }
+
+    /** {@inheritDoc} */
+    @Override protected void afterTest() throws Exception {
+        stopAllGrids();
+    }
+
+    /** */
+    @Test
+    public void testIgniteCacheClear() throws Exception {
+        startGrid(0);
+
+        checkIgniteCacheClear(grid(0));
+    }
+
+    /** */
+    @Test
+    public void testIgniteCacheClearOnClientNode() throws Exception {
+        startGrid(0);
+
+        startClientGrid(1);
+
+        checkIgniteCacheClear(grid(1));
+    }
+
+    /**
+     * It should throw exception.
+     *
+     * @param ignite Ignite.
+     */
+    private void checkIgniteCacheClear(IgniteEx ignite) {
+        IgniteCache<Object, Object> cache = ignite.cache(DEFAULT_CACHE_NAME);
+
+        cache.put(1, 1);
+
+        assertThrows(null,
+            () -> doInTransaction(ignite, () -> {
+                cache.put(2, 2);
+
+                cache.clear();
```

Comment:
cache.clear(2) also doesn't support transactions. You can check it with similar test 

```
        IgniteCache<Object, Object> cache = ignite.cache(DEFAULT_CACHE_NAME);

        cache.put(1, 1);

        doInTransaction(ignite, () -> {
                cache.put(2, 2);

                cache.clear(2);

                return null;
            });

        assertTrue(cache.containsKey(1));
        assertFalse(cache.containsKey(2));
```

Reply:
Let's disallow calling all clear() methods in transactions.